Bk 4320 update tree component

brikiernan opened the pull request:
Draft pr to review onConnectedCallback usage and initial style changes.

brikiernan commented on packages/web-components/src/components/rux-tree-node/rux-tree-node.tsx:
```diff
@@ -275,10 +285,16 @@ export class RuxTreeNode {
                         {this._hasChildren && (
-                            <i
+                            <rux-icon
+                                class="icon"
```
I changed the class and updated the arrow closed/expanded logic

markacianfrani commented:
looking really good. can we:

1. NIT - rename nowrap custom prop to --white-space? so its more intuitive as a developer that the options are wrap, nowrap, etc. we'll just need to add some documentation in the tree story around that. 
2. NIT - rename the left/right slots to prefix/suffix to be consistent with what input calls them 

3. There's a little bit of jank happening when you expand/collapse a node

https://user-images.githubusercontent.com/1091342/187769063-eae669c5-3cb0-435e-afb6-29481cfd74b5.mov

markacianfrani commented:
4. We can actually remove the code that swaps the light theme status icons on selected. Previously, we needed to do that bc the selected bg color didnt have enough contrast with the status symbol icons but the new color doesnt have that issue. If you notice right now, when a node with a status icon is selected, the icons change to have a darker border.

brikiernan commented:
The four changes have been made as detailed.

markacianfrani commented on packages/web-components/src/components/rux-tree-node/rux-tree-node.tsx:
```diff
@@ -253,34 +256,56 @@ export class RuxTreeNode {
+                        )}
+                        {hasPrefix && (
+                            <span class="prefix">
+                                <slot name="prefix"></slot>
```
since the slot is being conditionally rendered, we can't add slotted content after the component renders. 

For example, we should be able to do: 

```
<body>
        <rux-tree-node style="width: 300px;">whatup</rux-tree-node>
        <button id="add">add a prefix</button>
        <script>
            add.addEventListener('click', () => {
                const el = document.querySelector('rux-tree-node')
                const icon = document.createElement('rux-status')
                icon.setAttribute('status', 'critical')
                icon.setAttribute('slot', 'prefix')
                el.appendChild(icon)
            })
        </script>
    </body>
```

brikiernan commented:
@markacianfrani I updated the suffix and prefix slots to not conditionally render.

I'm not sure how that conflicts with the checkbox files got it here or how to resolve them.

markacianfrani commented on packages/web-components/src/components/rux-tree-node/rux-tree-node.tsx:
```diff
@@ -67,7 +83,7 @@ export class RuxTreeNode {
     }
 
-    @Listen('keydown', { passive: true })
+    @Listen('keydown', { passive: false })
```
do you know what the issue with this was?

brikiernan replied:
Yes, if you have any scroll, and press the right or down key arrow, you do what that key should do in the component and scroll at the same time.

Basically if it's set to true the e.preventDefault does not work and you'll get an error saying something like "preventDefault won't work on passive events"

kiley-mitti commented:
Added the new tree parts for 7.0 here with Brian's excellent help! :)

markacianfrani commented on packages/web-components/src/components/rux-checkbox/rux-checkbox.scss:
```diff
@@ -46,6 +46,7 @@
     input[type='checkbox'] {
```
can you remove the checkbox PR stuff from this PR please?

brikiernan commented:
@markacianfrani I removed the "wrap" prop and adjusted the css. Also, should I remove all the code I added in the test html file for rux-tree on this is ready to go?

markacianfrani commented on packages/web-components/src/components/rux-tree-node/rux-tree-node.scss:
```diff
@@ -1,67 +1,34 @@
     font-family: var(--font-body-1-font-family);
-    font-size: var(--font-body-1-font-size);
-    font-weight: var(--font-body-1-font-weight);
+    font-size: var(--font-size-base);
```
any reason you're not using `--font-control-body-1-font-size`? and `--font-control-body-1-line-height`

brikiernan replied:
I think I thought that was the correct token to use now, I'll change the size token back and I'll add the line-height on that class also
